src/utils/changeChain.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  internalAddressesPerNetwork,
+  ValidNetworkNameTypes,
+  ChainAddresses,
+} from "../utils/contracts/contractAddresses/internalAddresses";
+import { injectedConnector } from "../hooks/injectedConnectors";
+
+declare class RequestParameters {
+  chainId: string;
+  chainName?: string | null;
+  rpcUrls?: (string | undefined)[];
+  blockExplorerUrls?: (string | undefined)[];
+  nativeCurrency?: {
+    name: string;
+    symbol: string;
+    decimals: number;
+  };
+}
+
+function createRequestArguments(
+  method: string,
+  chain: ChainAddresses
+): {
+  id: number;
+  jsonrpc: string;
+  method: string;
+  params: RequestParameters[];
+} {
+  const params: RequestParameters[] =
+    method === "wallet_switchEthereumChain"
+      ? [
+          {
+            chainId: `0x${chain.chainId.toString(16)}`,
+          },
+        ]
+      : [
+          {
+            chainId: `0x${chain.chainId.toString(16)}`,
+            chainName: chain.chainName,
+            rpcUrls: [chain.rpcUrl],
+            nativeCurrency: {
+              name: chain.symbol,
+              symbol: chain.symbol,
+              decimals: 18,
+            },
+            blockExplorerUrls: [chain.explorer],
+          },
+        ];
+  return {
+    id: 1,
+    jsonrpc: "2.0",
+    method: method,
+    params: params,
+  };
+}
+
+// Only supported with Metamask
+export function changeChain(chainName: ValidNetworkNameTypes) {
+  const chain = internalAddressesPerNetwork[chainName];
+  injectedConnector.getProvider().then(async (provider: any) => {
+    await provider
+      .request(createRequestArguments("wallet_switchEthereumChain", chain))
+      .catch(
+        await provider.request(
+          createRequestArguments("wallet_addEthereumChain", chain)
+        )
+      );
+  });
+}
```

Comment:
This should be async instead of use of `.then` etcetera, and should return a `Promise` so it can be awaited for its results.
I'll be pushing a commit to your branch that revises this whole file.

Reply:
Did this myself in my commit, check it out and resolve this comment when you see it.